dataprofiler/profilers/profile_builder.py

```diff
@@ -1497,23 +1497,81 @@ def save(self, filepath=None):
         self._save_helper(filepath, data_dict)
 
 
-def Profiler(data, samples_per_update=None, min_true_samples=0, options=None):
-    """
-    Wrapper function for instantiating Structured and Unstructured Profilers
-
-    :param data: Data to be profiled
-    :type data: Data class object
-    :param samples_per_update: Number of samples to use in generating profile
-    :type samples_per_update: int
-    :param min_true_samples: Minimum number of samples required for the profiler
-    :type min_true_samples: int
-    :param options: Options for the profiler.
-    :type options: ProfilerOptions Object
-    :return: BaseProfiler
-    """
-    # Will want to add 'profiler_type' parameter similar to 'labeler_type'
-    # for specifying structured/unstructured profiler
-
-    # TODO: Add support for creating unstructured profilers via this wrapper
-    return StructuredProfiler(data, samples_per_update, min_true_samples,
-                              options)
+class Profiler(object):
+
+    def __new__(cls, data, samples_per_update=None, min_true_samples=0,
+                options=None, profiler_type=None):
+        """
+        Factory class for instantiating Structured and Unstructured Profilers
+
+        :param data: Data to be profiled
+        :type data: Data class object
+        :param samples_per_update: Number of samples to use to generate profile
+        :type samples_per_update: int
+        :param min_true_samples: Min number of samples required for the profiler
+        :type min_true_samples: int
+        :param options: Options for the profiler.
+        :type options: ProfilerOptions Object
+        :param profiler_type: Type of Profiler ("structured"/"unstructured")
+        :type profiler_type: str
+        :return: BaseProfiler
+        """
+        if profiler_type and profiler_type not in ["structured", "unstructured"]:
+            raise ValueError("Must specify 'profiler_type' to be 'structured' "
+                             "or 'unstructured'.")
+
+        if profiler_type is None:
+            # If type unspecified, and data is Data object, use is_structured
```

Comment:
if you set here: `profiler_type = "structured"` then you can remove the checks for the other cases, warnings would still need to occur like in the `len(data)` case, but less to check then.